airflow/utils/log/secrets_masker.py

```diff
@@ -144,6 +144,13 @@ def _record_attrs_to_ignore(self) -> Iterable[str]:
         )
         return frozenset(record.__dict__).difference({'msg', 'args'})
 
+    def _redact_exception_with_context(self, exception):
+        exception.args = (self.redact(v) for v in exception.args)
+        if exception.__context__:
+            self._redact_exception_with_context(exception.__context__)
+        if exception.__cause__:
```

Comment:
```suggestion
        if exception.__cause__ and exception.__cause__ is not exception.__context__:
```

to avoid processing it twice?

Reply:
Yep. Super-duper-optimised :)